File: app/src/main/java/com/thesis/admin/Administrators.java
package com.thesis.admin;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.os.Bundle;
import android.view.Menu;
import android.view.MenuItem;
import android.widget.SearchView;

import com.firebase.ui.database.FirebaseRecyclerOptions;
import com.google.firebase.database.FirebaseDatabase;
import com.thesis.admin.data.Clients;

public class Administrators extends AppCompatActivity {

    RecyclerView recyclerView;
    adminadapter adminadapter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_administrators);
        getSupportActionBar().setDisplayHomeAsUpEnabled(true);



        recyclerView = findViewById(R.id.clientlisthistory);
        recyclerView.setLayoutManager(new LinearLayoutManager(this));

        FirebaseRecyclerOptions<Adminis> options =
                new FirebaseRecyclerOptions.Builder<Adminis>()
                        .setQuery(FirebaseDatabase.getInstance().getReference("AdminDB"), Adminis.class)
                        .build();

        adminadapter = new adminadapter(options);
        recyclerView.setAdapter(adminadapter);
    }
    @Override
    protected void onStart() {
        super.onStart();
        adminadapter.startListening();
    }

    @Override
    protected void onStop() {
        super.onStop();
        adminadapter.stopListening();
    }

    @Override
    public  boolean onCreateOptionsMenu(Menu menu) {

        getMenuInflater().inflate(R.menu.search, menu);
        MenuItem item = menu.findItem(R.id.searchbar);
        SearchView searchView = (SearchView)item.getActionView();

        searchView.setOnQueryTextListener(new SearchView.OnQueryTextListener() {
            @Override
            public boolean onQueryTextSubmit(String query) {
                textsearch(query);
                return false;
            }

            @Override
            public boolean onQueryTextChange(String query) {
                textsearch(query);
                return false;
            }
        });
        return super.onCreateOptionsMenu(menu);
    }

    private void textsearch(String str) {
        FirebaseRecyclerOptions<Adminis> options =
                new FirebaseRecyclerOptions.Builder<Adminis>()
                        .setQuery(FirebaseDatabase.getInstance().getReference("AdminDB").orderByChild("AdminName").startAt(str).endAt(str+"~"), Adminis.class)
                        .build();

        adminadapter = new adminadapter(options);
        adminadapter.startListening();
        recyclerView.setAdapter(adminadapter);
    }
}

File: app/src/main/java/com/thesis/admin/Administrators.kt
package com.thesis.admin

import android.annotation.TargetApi
import android.content.Intent
import android.os.Build
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.Window
import android.view.WindowManager
import android.widget.AutoCompleteTextView
import android.widget.Button
import android.widget.EditText
import android.widget.TextView
import androidx.annotation.RequiresApi
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.google.firebase.auth.FirebaseAuth
import com.google.firebase.database.*

class Administrators : AppCompatActivity() {

    private lateinit var dbref :DatabaseReference
    private lateinit var clienthistory : RecyclerView
    private lateinit var clientArrayList : ArrayList<Adminis>
    private lateinit var clientsearch: AutoCompleteTextView

    @TargetApi(Build.VERSION_CODES.LOLLIPOP)
    @RequiresApi(api = Build.VERSION_CODES.JELLY_BEAN)
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_administrators)
        getWindow().setFlags(
                WindowManager.LayoutParams.FLAG_FULLSCREEN,
                WindowManager.LayoutParams.FLAG_FULLSCREEN
        );
        getSupportActionBar()?.hide();


        clienthistory = findViewById(R.id.clientlisthistory)
        clienthistory.layoutManager = LinearLayoutManager(this,)
        clienthistory.setHasFixedSize(true)
        clientArrayList = arrayListOf()


        loadProfile()
    }

        private fun loadProfile() {

            clientArrayList = arrayListOf()
                dbref = FirebaseDatabase.getInstance().getReference("AdminDB")
                dbref.addValueEventListener(object :ValueEventListener{


                    override fun onDataChange(snapshot: DataSnapshot) {
                        if (snapshot.exists()){
                            for(userSnapshot in snapshot.children){
                                val users = userSnapshot.getValue(Adminis::class.java)
                                clientArrayList.add(users!!)
                            }
                        }
                        clienthistory.adapter =adminadapter(clientArrayList)
                    }

                    override fun onCancelled(error: DatabaseError) {
                        TODO("Not yet implemented")
                    }

                })


            val back1 = findViewById<TextView>(R.id.backtohome)
            back1.setOnClickListener {
                startActivity(Intent(this@Administrators, Home_menu::class.java))
                finish()
            }
            }
        }


